stats/stats.go

```diff
@@ -393,21 +390,56 @@ func (h *Histogram) Clone() *Histogram {
 // CopyFrom sets the content of this object to a copy of the src.
 func (h *Histogram) CopyFrom(src *Histogram) {
 	h.Counter = src.Counter
-	// we don't copy offset/divider as this assumes compatible src/dest
-	for i := 0; i < len(h.Hdata); i++ {
-		h.Hdata[i] += src.Hdata[i]
+	h.CopyHDataFrom(src)
+}
+
+func (h *Histogram) CopyHDataFrom(src * Histogram)  {
+	if h.Divider == src.Divider && h.Offset == src.Offset {
+		for i := 0; i < len(h.Hdata); i++ {
+			h.Hdata[i] += src.Hdata[i]
+		}
+		return;
 	}
+	//Scale values are not equal for histograms.
+	//Need to receive record values from src and add them to h according to h's offset and divider
+	var scaledValue float64
+	for i := 0; i < len(src.Hdata); i++ {
```

Comment:
this may be easier and more correct if you use Export() on src first

or make a function on how to return the interval for each bucket that both export and this can use

you want to use at least the mid point of the range as the value (though it's not quite the best that can be done when re-bining) there are some subtleties around min and max too (ie if 1 bucket is [min, x] and has a count of 1 the value to use is min, not (min+x)/2))

Reply:
As you mention I used Export method on src to receive histogramData. And I processed its Datas with formula that you mention on above.